### PR TITLE
target addresses of listening ports and add control for troubleshooting

### PR DESCRIPTION
**Pull request by alexpop**
Listening ports can be bound on any interface available on the node. If this is the case, the ssl resource needs to use that IP address and not localhost or target IP.

### CODE REVIEW

File: controls/ssl_test.rb [hunk 1]
```diff
@@ -18,16 +18,38 @@
...
+# Method used to build the params for the ssl resource
+# Uses the port's listening address unless it's localhost or any. In this case, the ssl resource will use the target's IP/host
+target_hash = lambda { |s|
+  params = { port: s.port }
+  # params[:timeout] = 8
+  # params[:retries] = 1
+  invalid_addrs = ['127.0.0.1', '0.0.0.0', '::1', '::']
+  params[:host] = s.address unless invalid_addrs.include?(s.address)
+  return params
+}
```
Comment (arlimus) [inline controls/ssl_test.rb]: Maybe as a different suggestion: (still not happy here)

```
invalid_addrs = %w{
  127.0.0.1
  0.0.0.0
  ::1
  ::
}

tcpports = ...
enabled_sockets = ...
ssl_list = enabled_sockets.map do |s|
  params = { port: s.port }
  params[:host] = s.address unless invalid_addrs.include?(s.address)
  [s, ssl(params)]
end

control 'tls1.2' do
   title 'Run TLS 1.2 whenever SSL is active on a port'
   impact 0.5
 
   ssl_list.each do |socket, ssl_socket|
     # create a description
     proc_desc = "on node == #{command('hostname').stdout.strip} running #{socket.process.inspect} (#{socket.pid})"
     describe ssl_socket.protocols('tls1.2') do
       it(proc_desc) { should be_enabled }
       it { should be_enabled }
     end
   end
end
```

**Comment by alexpop** [comment]
Thanks for the feedback Dom.
I updated my branch to handle the host without the lambda method.

Also getting the hostname only once instead of 6 times per port.

I've also made a few more cosmetic changes.

I would keep the debugging control until we iron out all the edge cases.

**Comment by alexpop** [comment]
screen shot 2016-10-10 at 6 24 39 pm

^ sample scan with this profile

**Comment by chris-rock** [comment]
@arlimus Could you please re-review @alexpop updates?

**Comment by chris-rock** [comment]
Thank you @alexpop

**Pull request merged**